### ISSUE TITLE
CSVDataStore and Metadata Questions

### ISSUE THREAD

**Issue by suni9915** [nilmtk/nilm_metadata#29]
I'd like to use NILMTK for my own input data from the smart plug.  I've read some of the document but still confused about the CSVDataStore and Metadata.

My input data is in CSV file format and I intend to match it to the REDD dataset format and use the REDD converter to convert it to HDF5 format.  Do I have to implement the CSVDatastore before running the REDD converter?  And about the metadata that I have to build for my own data, is there any tool to help  building it?   Thank you very much.

**Comment by nipunbatra** [comment]
>Do I have to implement the CSVDatastore before running the REDD converter? 

No. See: https://github.com/nilmtk/nilmtk/blob/master/nilmtk/dataset_converters/redd/convert_redd.py

>And about the metadata that I have to build for my own data, is there any tool to help building it? 

We don't have a tool yet. However, it shouldn't take long to write the metadata by hand. 
How many buildings/meters do you have?

**Comment by tampubolon** [comment]
Hello Nipunbatra

I have the same problem. I have 1-2 Buildings.
